assets-sync.js

```diff
@@ -1407,15 +1407,15 @@ class WorldMigration {
                 return localPath;
             }
         } else {
-            const asset = this.assets.get(decodeURI(name));
+            const asset = this.assets.get(ForgeAssetSync.sanitizePath(decodeURIComponent(name)));
```

Comment:
The reference to the asset in the Foundry world, i.e. `name`, is URI encoded at this point, but not sanitized with `sanitizePath()`.
The local assets that have been synced from the assets library are in `this.assets`. They've been sanitized with `sanitizePath()`, but they're not URI encoded.

To make sure that we check if the asset which is referenced by `name` properly tries to find its counterpart in `this.assets`, we need to decodeURI the name, and then sanitize it. At which point, both strings will have undergone the same treatment.

I changed to using `decodeURIComponent` instead of `decodeURI` because `decodeURI` excludes specific characters like the `:` character, which means that what's sent to the `sanitizePath()` function as input could still be URI encoded and the strings won't match after being sanitized.

For example: 
![image](https://github.com/user-attachments/assets/57e28879-66eb-45c2-a247-d922bf7a711d)